File: hospitalManagement/Medicine.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hospitalManagement
{
    internal class Medicine
    {
        //Field
        private string id;
        // Infor field
        private string name;
        private string description;
        private string type;
        private string dosage;
        private string storage;
        private float weight;
        private string serialNumber;
        private ComeAndLeaveDate expiryDate;


        //  Management field
        // Dynamic field
        // Properties
        public string Id { get => id; set => id = value; }
        public string Name { get => name; set => name = value; }
        public string Description { get => description; set => description = value; }
        public string Type { get => type; set => type = value; }
        public string Dosage { get => dosage; set => dosage = value; }
        public string Storage { get => storage; set => storage = value; }
        public float Weight { get => weight; set => weight = value; }
        public string SerialNumber { get => serialNumber; set => serialNumber = value; }
        public ComeAndLeaveDate ExpiryDate { get => expiryDate; set => expiryDate = value; }

        // Constructors
        public Medicine()
        {
            expiryDate = new ComeAndLeaveDate();
        }
        public Medicine(string id
            , string name
            , string description
            , string type
            , string dosage
            , string storage
            , float weight
            , string serialNumber
            , ComeAndLeaveDate expiryDate)
        {
            this.id = id;
            this.name = name;
            this.description = description;
            this.type = type;
            this.dosage = dosage;
            this.storage = storage;
            this.weight = weight;
            this.serialNumber = serialNumber;
            this.expiryDate = expiryDate;
        }
        // Destructors
        ~Medicine() { }
        // Methods
        // in, output
        public void Input()
        {
            Console.Write("Id: ");
            id = Console.ReadLine();
            Console.Write("Name: ");
            name = Console.ReadLine();
            Console.Write("Description: ");
            description = Console.ReadLine();
            Console.Write("Type: ");
            type = Console.ReadLine();
            Console.Write("Dosage: ");
            dosage = Console.ReadLine();
            Console.Write("Storage: ");
            storage = Console.ReadLine();
            Console.Write("Weight: ");
            weight = float.Parse(Console.ReadLine());
            Console.Write("The serial number: ");
            serialNumber = Console.ReadLine();
            Console.WriteLine("The expiry date");
            expiryDate.Input();
        }
        public void Output()
        {
            Console.WriteLine($"Id: {id}");
            Console.WriteLine($"Name: {name}");
            Console.WriteLine($"Description: {description}");
            Console.WriteLine($"Type: {type}");
            Console.WriteLine($"Dosage: {dosage}");
            Console.WriteLine($"Storage: {storage}");
            Console.WriteLine($"Weight: {weight}");
            Console.WriteLine($"SerialNumber: {serialNumber}");
            Console.WriteLine(expiryDate);
        }



        // General method

        // Other method


        // Overriding
        public override string ToString()
        => $"Id: {id}"
        + $"\nName: {name}"
        + $"\nDescription: {description}"
        + $"\nType: {type}"
        + $"\nDosage: {dosage}"
        + $"\nStorage: {storage}"
        + $"\nWeight: {weight}"
        + $"\nSerialNumber: {serialNumber}"
        + $"\n{expiryDate}";
    }
}

File: src/hospitalManagement/Medicine.java
package hospitalManagement;

import java.util.Scanner;

public class Medicine {
	//Field
    private String id;
    // Infor field
    private String name;
    private String description;
    private String type;
    private String dosage;
    private String storage;
    private float weight;
    private String serialNumber;
    private ComeAndLeaveDate expiryDate;


    //  Management field
    // Dynamic field
    // Properties
   

    public String getId() {
		return id;
	}
	public void setId(String id) {
		this.id = id;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public String getDescription() {
		return description;
	}
	public void setDescription(String description) {
		this.description = description;
	}
	public String getType() {
		return type;
	}
	public void setType(String type) {
		this.type = type;
	}
	public String getDosage() {
		return dosage;
	}
	public void setDosage(String dosage) {
		this.dosage = dosage;
	}
	public String getStorage() {
		return storage;
	}
	public void setStorage(String storage) {
		this.storage = storage;
	}
	public float getWeight() {
		return weight;
	}
	public void setWeight(float weight) {
		this.weight = weight;
	}
	public String getSerialNumber() {
		return serialNumber;
	}
	public void setSerialNumber(String serialNumber) {
		this.serialNumber = serialNumber;
	}
	public ComeAndLeaveDate getExpiryDate() {
		return expiryDate;
	}
	public void setExpiryDate(ComeAndLeaveDate expiryDate) {
		this.expiryDate = expiryDate;
	}
	
	// Constructors
    public Medicine()
    {
        expiryDate = new ComeAndLeaveDate();
    }
    public Medicine(String id
        , String name
        , String description
        , String type
        , String dosage
        , String storage
        , float weight
        , String serialNumber
        , ComeAndLeaveDate expiryDate)
    {
        this.id = id;
        this.name = name;
        this.description = description;
        this.type = type;
        this.dosage = dosage;
        this.storage = storage;
        this.weight = weight;
        this.serialNumber = serialNumber;
        this.expiryDate = expiryDate;
    }
    // Destructors
    
    // Methods
    // in, output
    public void Input()
    {
    	Scanner input = new Scanner(System.in);
        System.out.format("Id: ");
        id = input.nextLine();
        System.out.format("Name: ");
        name = input.nextLine();
        System.out.format("Description: ");
        description = input.nextLine();
        System.out.format("Type: ");
        type = input.nextLine();
        System.out.format("Dosage: ");
        dosage = input.nextLine();
        System.out.format("Storage: ");
        storage = input.nextLine();
        System.out.format("Weight: ");
        weight = input.nextInt();
        System.out.format("The serial number: ");
        serialNumber = input.nextLine();
        System.out.format("The expiry date: \n");
        expiryDate.Input();
        input.close();
    }
    public void Output()
    {                     
        System.out.format("Id: %s\n", id);
        System.out.format("Name: %s\n", name);
        System.out.format("Description: %s\n", description);
        System.out.format("Type: %s\n", type);
        System.out.format("Dosage: %s\n", dosage);
        System.out.format("Storage: %s\n", storage);
        System.out.format("Weight: %d", weight);
        System.out.format("SerialNumber: %s\n", serialNumber);
        expiryDate.Output();
    }



    // General method

    // Other method


    // Overriding
    public  String toString()
    {
    	return "Id: " + id
			    + "\nName: " + name
			    + "\nDescription: " + description
			    + "\nType: " + type
			    + "\nDosage: " + dosage
			    + "\nStorage: " + storage
			    + "\nWeight: " + weight
			    + "\nSerialNumber: " + serialNumber
			    + "\n" + expiryDate;
    }
}
